src/pages/ProfilesExplorerView/components/SceneByVariableRepeaterGrid/SceneByVariableRepeaterGrid.tsx

```diff
@@ -68,29 +72,58 @@ const DEFAULT_SORT_ITEMS_FN: SceneByVariableRepeaterGridState['sortItemsFn'] = f
   return 0;
 };
 
-export class SceneByVariableRepeaterGrid extends SceneObjectBase<SceneByVariableRepeaterGridState> {
-  static DEFAULT_LAYOUT = LayoutType.GRID;
-
-  static buildGridItemKey(item: GridItemData) {
-    return `grid-item-${item.index}-${item.value}`;
-  }
-
-  static limitNumberOfSeries = () => (source: Observable<DataFrame[]>) =>
+export const DATA_TRANSFORMATIONS = {
```

Comment:
Splitting data transformations by responsibility to compose them more easily, based on the panel type.